host/sgx/sgxload.c

```diff
@@ -27,9 +27,68 @@
 #include <openenclave/internal/utils.h>
 #include "../memalign.h"
 #include "../signkey.h"
+#include "cpuid.h"
 #include "enclave.h"
 #include "sgxmeasure.h"
 
+#if defined(__linux__)
+// If cpuid.01h:ECX.XSAVE[bit 26] is 1, the processor supports the XSAVE/XRSTOR
+// processor extended states feature, the XSETBV/XGETBV instructions, and XCR0
+static bool is_xcr0_supported(void)
+{
+    uint32_t eax, ebx, ecx, edx;
+
+    eax = ebx = ecx = edx = 0;
+    oe_get_cpuid(1, 0, &eax, &ebx, &ecx, &edx);
+
+    // Bit 26 of ECX indicates support for XGETBV and XCR0
+    if (!(ecx & (1 << 26)))
+        return false;
+
+    return true;
+}
+
+static uint64_t xgetbv(uint32_t index)
+{
+    uint32_t eax, edx;
+    eax = edx = 0;
+
+    if (is_xcr0_supported())
+    {
+        asm volatile(".byte 0x0f,0x01,0xd0" /* xgetbv */
```

Comment:
Please don't use inline assembly in host code; also implement the Windows equivalent and test it. 

I would also recommend following the existing pattern of splitting the implementation between linux and windows folders under host/sgx for clarity. i.e. define a header with `oe_get_xfrm()` and with separate .c files under the appropriate OS folder.

Minor note: probably clearer to use `is_xgetbv_supported` because its not clear to a dev without context that support for `xcr0` implies support for `xgetbv`/`xsetbv` and vice versa.

Reply:
ok, will incorporate

Reply:
Done

Reply:
Separate .c/.h files created for Linux in the host/sgx/linux folder.